server/src/main/java/org/opensearch/indices/tiering/TieringRequestValidator.java

```diff
@@ -0,0 +1,300 @@
+                tieringValidationResult.addToRejected(
+                    index,
+                    "index is already in the warm tier or the tiering is currently in progress for the index"
+                );
+                continue;
+            }
+            if (!validateRemoteStoreIndex(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is not backed up by the remote store");
+                continue;
+            }
+            if (!validateOpenIndex(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is closed");
+                continue;
+            }
+            if (!validateIndexHealth(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is red");
+                continue;
+            }
+            tieringValidationResult.addToAccepted(index);
+        }
+
+        final Tuple<Set<Index>, Set<Index>> eligibleNodesCapacity = validateEligibleNodesCapacity(
+            clusterInfo,
+            currentState,
+            tieringValidationResult.getAcceptedIndices()
+        );
+        tieringValidationResult.addToAccepted(eligibleNodesCapacity.v1());
+        tieringValidationResult.addToRejected(eligibleNodesCapacity.v2(), "insufficient node capacity");
+        logger.info(
+            "Successfully accepted indices for tiering are [{}], rejected indices are [{}]",
+            tieringValidationResult.getAcceptedIndices(),
+            tieringValidationResult.getRejectedIndices()
+        );
+
+        return tieringValidationResult;
+    }
+
+    /**
+     * Validates that there are eligible nodes with the search role in the current cluster state.
+     * (only for the dedicated case - to be removed later)
+     *
+     * @param currentState the current cluster state
+     * @param indexNames the names of the indices being validated
+     * @throws IllegalArgumentException if there are no eligible search nodes in the cluster
+     */
+    static void validateSearchNodes(final ClusterState currentState, final String indexNames) {
+        if (getEligibleNodes(currentState).isEmpty()) {
+            final String errorMsg = "Rejecting tiering request for indices ["
+                + indexNames
+                + "] because there are no nodes found with the search role";
+            logger.warn(errorMsg);
+            throw new IllegalArgumentException(errorMsg);
+        }
+    }
+
+    /**
+     * Validates that the specified index has the remote store setting enabled.
+     *
+     * @param state the current cluster state
+     * @param index the index to be validated
+     * @return true if the remote store setting is enabled for the index, false otherwise
+     */
+    static boolean validateRemoteStoreIndex(final ClusterState state, final Index index) {
+        return IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.get(state.metadata().getIndexSafe(index).getSettings());
+    }
```

Comment:
lets change it to `Boolean.FALSE == IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.get(state.metadata().getIndexSafe(index).getSettings())`.

Reply:
`Boolean.FALSE == IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.get(state.metadata().getIndexSafe(index).getSettings())` this is not the right check - it would return true when the setting is not present

also, if the setting is not present - it returns the default value which is false
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/Setting.java#L477

Reply:
sorry.. my bad.. i meant `Boolean.TRUE == IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.get(state.metadata().getIndexSafe(index).getSettings())`.